matlab/src/feather_reader.cc

```diff
@@ -177,32 +182,32 @@ Status FeatherReader::Open(const std::string& filename,
   *feather_reader = std::shared_ptr<FeatherReader>(new FeatherReader());
 
   // Open file with given filename as a ReadableFile.
-  std::shared_ptr<io::ReadableFile> readable_file(nullptr);
-
-  RETURN_NOT_OK(io::ReadableFile::Open(filename, &readable_file));
+  arrow::Result<std::shared_ptr<io::ReadableFile>> maybe_readable_file =
+      io::ReadableFile::Open(filename);
 
   // TableReader expects a RandomAccessFile.
-  std::shared_ptr<io::RandomAccessFile> random_access_file(readable_file);
+  ARROW_ASSIGN_OR_RAISE(std::shared_ptr<io::RandomAccessFile> random_access_file,
+                        maybe_readable_file);
```

Comment:
Do we need this cast?

I think that `arrow::ipc::feather::Reader::Open()` can accept `arrow::io::ReadableFile` because `arrow::io::ReadableFile` is a subclass of `arrow::io::RandomAccessFile`.

Reply:
You're right. We don't need this cast.